### PR TITLE
ci: fix periodic, add dependabot, support docs-only and update otel workflow

### PR DESCRIPTION
**Pull request by v1v**
### What

* Enable dependabot
* Split workflows in the ones for periodic runs and the ones for CI, to bypass https://github.com/orgs/community/discussions/66834
* Update opentelemetry workflow
* Enable docs-ci

### CODE REVIEW

File: .github/workflows/matrix.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
```
Comment (v1v) [inline .github/workflows/matrix.yml:1]: This workflow is reused, so there is no need to maintain the same step in two different workflows:
- test.yml
- periodic.yml

File: .github/workflows/test-docs.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+---
```
Comment (v1v) [inline .github/workflows/test-docs.yml:1]: This will solve CI checks when only doc changes

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -34,16 +29,6 @@ jobs:
         python-version: '3.7'
     - run: .ci/scripts/lint.sh
 
+  # Invokes the actual matrix workflow with the provided files.
```
Comment (v1v) [inline .github/workflows/test.yml:32]:
```suggestion
  # Invokes the actual matrix workflow
```

File: .github/workflows/periodic.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  # Run daily at midnight
+  schedule:
+    - cron: "0 0 * * *"
+
+permissions:
+  contents: read
+
+jobs:
+
+  # Invokes the actual matrix workflow with the provided files.
```
Comment (v1v) [inline .github/workflows/periodic.yml:13]:
```suggestion
  # Invokes the actual matrix workflow.
```

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -17,14 +14,12 @@ permissions:
 jobs:
```
Comment (v1v) [inline .github/workflows/test.yml:14]:
```suggestion

## Concurrency is only allowed in the main branch.
## So old builds, running for old commits within the same Pull Request, are cancelled
concurrency:
  group: ${{ github.workflow }}-${{ github.ref }}
  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}

jobs:
```

**Review by basepi — APPROVED** [review]

**Pull request merged**